File: apps/api/domain/book.go
package domain

import "time"

type Book struct {
    ID        string    `json:"id"`
    Title     string    `json:"title"`
    Author    string    `json:"author"`
    Price     float64   `json:"price"`
    Currency  string    `json:"currency"`
    Stock     int       `json:"stock"`
    CreatedAt time.Time `json:"created_at"`
    UpdatedAt time.Time `json:"updated_at"`
}

type CreateBookInput struct {
    Title    string  `json:"title"`
    Author   string  `json:"author"`
    Price    float64 `json:"price"`
    Currency string  `json:"currency"`
    Stock    int     `json:"stock"`
}

type UpdateBookInput struct {
    Title    string  `json:"title"`
    Author   string  `json:"author"`
    Price    float64 `json:"price"`
    Currency string  `json:"currency"`
    Stock    int     `json:"stock"`
}

type PurchaseInput struct {
    BookID     string `json:"book_id"`
    Quantity   int    `json:"quantity"`
    CustomerID string `json:"customer_id"`
}

File: frontend/src/types/book.ts
export type Book = {
  id: number;
  title: string;
  author: string;
  isbn: string;
  price: number;
  stock: number;
  description: string;
  publishedDate: string;
  createdAt: string;
  updatedAt: string;
};

export type BookFormValues = {
  title: string;
  author: string;
  isbn: string;
  price: string;
  stock: string;
  description: string;
  publishedDate: string;
};

